File: java/src/main/java/fr/arolla/Ticketing.java
package fr.arolla;

public class Ticketing {
  private final EventBus bus;
  private int numberOfSeatsInLastPrint;

  public Ticketing(EventBus bus) {
    this.bus = bus;
  }

  public void printTicket(int numSeats) {
    System.out.format("Printing ticket for %d seats\n", numSeats);
    numberOfSeatsInLastPrint = numSeats;
    bus.emit(new TicketPrinted(numSeats));
  }

  public String getLastTicket() {
    return String.format("Ticket for %d seats", numberOfSeatsInLastPrint);
  }

}

File: dotnet/Choreography/Ticketing.cs
using Choreography.Events;

namespace Choreography;

public class Ticketing : ISubscriber
{
    private readonly Context _context;

    public Ticketing(Context context)
    {
        _context = context;
        var bus = _context.Bus();
        bus.Subscribe(this);
    }


    public void OnEvent(IEvent e)
    {
        if (e is CapacityUpdated capacityUpdated)
        {
            var numSeats = capacityUpdated.BookedSeats;
            PrintTickets(numSeats);
        }
    }

    public void PrintTickets(int numSeats)
    {
        _context.Logger().Log($"Printing tickets for {numSeats} seats");
        _context.Bus().Emit(new TicketPrinted());
    }
}